File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/job/UnmodifiableTask.java
package com.sos.scheduler.engine.kernel.job;

import com.sos.scheduler.engine.cplusplus.runtime.annotation.ForCpp;
import com.sos.scheduler.engine.data.job.TaskId;
import com.sos.scheduler.engine.eventbus.EventSource;
import com.sos.scheduler.engine.kernel.order.UnmodifiableOrder;

@ForCpp
public interface UnmodifiableTask extends EventSource {
	TaskId getId();
    UnmodifiableJob getJob();
    UnmodifiableOrder getOrderOrNull();
}

File: engine-kernel/src/main/scala/com/sos/scheduler/engine/kernel/job/UnmodifiableTask.scala
package com.sos.scheduler.engine.kernel.job

import com.sos.scheduler.engine.cplusplus.runtime.annotation.ForCpp
import com.sos.scheduler.engine.data.job.TaskId
import com.sos.scheduler.engine.eventbus.EventSource
import com.sos.scheduler.engine.kernel.order.UnmodifiableOrder

@ForCpp
trait UnmodifiableTask extends EventSource {

  private[kernel] def job: UnmodifiableJob

  private[kernel] def orderOption: Option[UnmodifiableOrder]

  private[kernel] def parameterValue(name: String): String
}
